packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -206,10 +206,10 @@ export class TokenBalancesController extends BaseController<
         newContractBalances[address] = toHex(
           await this.#getERC20BalanceOf(address, selectedAddress),
```

Comment:
Just a thought:

Would be possible to improve the performance of this function?


Doing for example something like this:

```
async updateBalances() {
  if (this.#disabled) {
    return;
  }

  const newContractBalances: ContractBalances = {};
  const balancePromises = this.#tokens.map(token => {
    const { address } = token;
    const { selectedAddress } = this.messagingSystem.call(
      'PreferencesController:getState',
    );

    // Wrap the balance fetching and error handling in a single promise
    return this.#getERC20BalanceOf(address, selectedAddress)
      .then(balance => {
        newContractBalances[address] = toHex(balance);
        token.balanceError = null;
      })
      .catch(error => {
        newContractBalances[address] = toHex(0);
        token.balanceError = error;
      });
  });

  // Wait for all balance fetching and processing to complete
  await Promise.all(balancePromises);

  this.update((state) => {
    state.contractBalances = newContractBalances;
  });
}
```